commit 392248a8898f374939fdbf24d44d958e73e891d3 (refs/changes/31/2231/3)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-07-18T22:51:39+00:00 (2 years, 3 months ago)
    
    VOLAPI-56 want log archiving setup with logadm for volapi-updater
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Approved by: Josh Wilsdon <jwilsdon@joyent.com>

diff --git a/.gitignore b/.gitignore
index 8ed9822..b6d0e83 100644
--- a/.gitignore
+++ b/.gitignore
@@ -6,5 +6,5 @@ docs/*.html
 cscope.in.out
 cscope.po.out
 cscope.out
-smf/manifests/volapi.xml
+smf/manifests/volapi-server.xml
 smf/manifests/volapi-updater.xml
diff --git a/Makefile b/Makefile
index 0661f3c..7205cea 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2016, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 #
@@ -24,7 +24,7 @@ JSL_FILES_NODE	 = $(JS_FILES)
 JSSTYLE_FILES	 = $(JS_FILES)
 JSSTYLE_FLAGS	 = -f tools/jsstyle.conf
 SERVICE_NAME     = volapi
-SMF_MANIFESTS_IN = smf/manifests/$(SERVICE_NAME).xml.in smf/manifests/$(SERVICE_NAME)-updater.xml.in
+SMF_MANIFESTS_IN = smf/manifests/$(SERVICE_NAME)-server.xml.in smf/manifests/$(SERVICE_NAME)-updater.xml.in
 
 NODE_PREBUILT_VERSION=v4.6.1
 
diff --git a/boot/configure.sh b/boot/configure.sh
index 1540c15..1b7b05b 100755
--- a/boot/configure.sh
+++ b/boot/configure.sh
@@ -7,7 +7,7 @@
 #
 
 #
-# Copyright (c) 2016, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
@@ -16,11 +16,11 @@ set -o xtrace
 service_name='volapi'
 
 echo "Updating SMF manifests"
-$(/opt/local/bin/gsed -i"" -e "s/@@PREFIX@@/\/opt\/smartdc\/${service_name}/g" /opt/smartdc/${service_name}/smf/manifests/${service_name}.xml)
+$(/opt/local/bin/gsed -i"" -e "s/@@PREFIX@@/\/opt\/smartdc\/${service_name}/g" /opt/smartdc/${service_name}/smf/manifests/${service_name}-server.xml)
 $(/opt/local/bin/gsed -i"" -e "s/@@PREFIX@@/\/opt\/smartdc\/${service_name}/g" /opt/smartdc/${service_name}/smf/manifests/${service_name}-updater.xml)
 
-echo "Importing ${service_name}.xml"
-/usr/sbin/svccfg import /opt/smartdc/${service_name}/smf/manifests/${service_name}.xml
+echo "Importing ${service_name}-server.xml"
+/usr/sbin/svccfg import /opt/smartdc/${service_name}/smf/manifests/${service_name}-server.xml
 
 echo "Importing ${service_name}-updater.xml"
 /usr/sbin/svccfg import /opt/smartdc/${service_name}/smf/manifests/${service_name}-updater.xml
diff --git a/boot/setup.sh b/boot/setup.sh
index f50aa2e..2f87673 100755
--- a/boot/setup.sh
+++ b/boot/setup.sh
@@ -7,7 +7,7 @@
 #
 
 #
-# Copyright (c) 2016, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
 export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
@@ -19,15 +19,15 @@ role=volapi
 
 # Include common utility functions (then run the boilerplate)
 source /opt/smartdc/boot/lib/util.sh
-CONFIG_AGENT_LOCAL_MANIFESTS_DIRS=/opt/smartdc/$role
+CONFIG_AGENT_LOCAL_MANIFESTS_DIRS=/opt/smartdc/${role}
 sdc_common_setup
 
 # Cookie to identify this as a SmartDC zone and its role
-mkdir -p /var/smartdc/$role
+mkdir -p /var/smartdc/${role}
 
 # Install VOLAPI
-mkdir -p /opt/smartdc/${role}
-chown -R nobody:nobody /opt/smartdc/${role}
+mkdir -p /opt/smartdc/$role
+chown -R nobody:nobody /opt/smartdc/$role
 
 # Add build/node/bin and node_modules/.bin to PATH
 echo "" >>/root/.profile
@@ -37,7 +37,8 @@ echo "Adding log rotation"
 # Log rotation.
 sdc_log_rotation_add config-agent /var/svc/log/*config-agent*.log 1g
 sdc_log_rotation_add registrar /var/svc/log/*registrar*.log 1g
-sdc_log_rotation_add $role /var/svc/log/*$role*.log 1g
+sdc_log_rotation_add volapi-server /var/svc/log/*volapi-server*.log 1g
+sdc_log_rotation_add volapi-updater /var/svc/log/*volapi-updater*.log 1g
 sdc_log_rotation_setup_end
 
 # All done, run boilerplate end-of-setup
diff --git a/sapi_manifests/volapi/manifest.json b/sapi_manifests/volapi/manifest.json
index 2a728b2..cf96ea0 100644
--- a/sapi_manifests/volapi/manifest.json
+++ b/sapi_manifests/volapi/manifest.json
@@ -1,5 +1,5 @@
 {
 	"name": "volapi",
 	"path": "/opt/smartdc/volapi/config.json",
-	"post_cmd": "/usr/sbin/svcadm restart volapi"
+	"post_cmd": "/usr/sbin/svcadm restart volapi-updater && /usr/sbin/svcadm restart volapi-server"
 }
diff --git a/smf/manifests/volapi-server.xml.in b/smf/manifests/volapi-server.xml.in
new file mode 100644
index 0000000..6d359ad
--- /dev/null
+++ b/smf/manifests/volapi-server.xml.in
@@ -0,0 +1,55 @@
+<?xml version="1.0"?>
+<!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
+<!--
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+-->
+
+<!--
+    Copyright (c) 2017, Joyent, Inc.
+-->
+
+<service_bundle type="manifest" name="volapi">
+    <service name="smartdc/application/volapi-server" type="service" version="1">
+
+    <dependency name="network" grouping="require_all" restart_on="error" type="service">
+        <service_fmri value="svc:/network/physical" />
+    </dependency>
+
+    <dependency name="filesystem" grouping="require_all" restart_on="error" type="service">
+        <service_fmri value="svc:/system/filesystem/local" />
+    </dependency>
+
+    <dependency name="mdata"
+                grouping="require_all"
+                restart_on="none"
+                type="service">
+        <service_fmri value="svc:/smartdc/mdata:execute" />
+    </dependency>
+
+    <dependency name="config-agent"
+                grouping="optional_all"
+                restart_on="none"
+                type="service">
+        <service_fmri value="svc:/smartdc/application/config-agent" />
+    </dependency>
+
+    <exec_method
+        type="method"
+        name="start"
+        exec="@@NODE@@ --abort_on_uncaught_exception @@PREFIX@@/server.js &amp;"
+        timeout_seconds="30" />
+
+    <exec_method type="method" name="stop" exec=":kill" timeout_seconds="30" />
+
+    <instance name="default" enabled="true" />
+
+    <stability value='Unstable' />
+
+    <template>
+        <common_name><loctext xml:lang="C">SDC Volumes API</loctext></common_name>
+    </template>
+
+    </service>
+</service_bundle>
diff --git a/smf/manifests/volapi-updater.xml.in b/smf/manifests/volapi-updater.xml.in
index ca71b1e..f533fdd 100644
--- a/smf/manifests/volapi-updater.xml.in
+++ b/smf/manifests/volapi-updater.xml.in
@@ -7,39 +7,48 @@
 -->
 
 <!--
-    Copyright (c) 2016, Joyent, Inc.
+    Copyright (c) 2017, Joyent, Inc.
 -->
 
-<!--
-  This is a sample SMF manifest. If you want to actually use it, you'll
-  need to update this block comment and replace @@PREFIX@@ in the *generated*
-  file with the path to whereever this package gets installed.
- -->
+<service_bundle type="manifest" name="volapi">
+    <service name="smartdc/application/volapi-updater" type="service" version="1">
+
+    <dependency name="network" grouping="require_all" restart_on="error" type="service">
+        <service_fmri value="svc:/network/physical" />
+    </dependency>
+
+    <dependency name="filesystem" grouping="require_all" restart_on="error" type="service">
+        <service_fmri value="svc:/system/filesystem/local" />
+    </dependency>
 
-<service_bundle type="manifest" name="smartdc-ca-castashsvc">
-<service name="smartdc/application/volapi-updater" type="service" version="1">
+    <dependency name="mdata"
+                grouping="require_all"
+                restart_on="none"
+                type="service">
+        <service_fmri value="svc:/smartdc/mdata:execute" />
+    </dependency>
 
-<dependency name="network" grouping="require_all" restart_on="error" type="service">
-	<service_fmri value="svc:/network/physical" />
-</dependency>
-<dependency name="filesystem" grouping="require_all" restart_on="error" type="service">
-	<service_fmri value="svc:/system/filesystem/local" />
-</dependency>
+    <dependency name="config-agent"
+                grouping="optional_all"
+                restart_on="none"
+                type="service">
+        <service_fmri value="svc:/smartdc/application/config-agent" />
+    </dependency>
 
-<exec_method
-    type="method"
-    name="start"
-    exec="@@NODE@@ --abort_on_uncaught_exception @@PREFIX@@/volapi-updater.js &amp;"
-    timeout_seconds="30" />
-<exec_method type="method" name="stop" exec=":kill" timeout_seconds="30" />
+    <exec_method
+        type="method"
+        name="start"
+        exec="@@NODE@@ --abort_on_uncaught_exception @@PREFIX@@/volapi-updater.js &amp;"
+        timeout_seconds="30" />
+    <exec_method type="method" name="stop" exec=":kill" timeout_seconds="30" />
 
-<instance name="default" enabled="true" />
+    <instance name="default" enabled="true" />
 
-<stability value='Unstable' />
+    <stability value='Unstable' />
 
-<template>
-	<common_name><loctext xml:lang="C">SDC Volumes API</loctext></common_name>
-</template>
+    <template>
+        <common_name><loctext xml:lang="C">SDC Volumes API</loctext></common_name>
+    </template>
 
-</service>
+    </service>
 </service_bundle>
diff --git a/smf/manifests/volapi.xml.in b/smf/manifests/volapi.xml.in
deleted file mode 100644
index 14533f9..0000000
--- a/smf/manifests/volapi.xml.in
+++ /dev/null
@@ -1,45 +0,0 @@
-<?xml version="1.0"?>
-<!DOCTYPE service_bundle SYSTEM "/usr/share/lib/xml/dtd/service_bundle.dtd.1">
-<!--
-    This Source Code Form is subject to the terms of the Mozilla Public
-    License, v. 2.0. If a copy of the MPL was not distributed with this
-    file, You can obtain one at http://mozilla.org/MPL/2.0/.
--->
-
-<!--
-    Copyright (c) 2016, Joyent, Inc.
--->
-
-<!--
-  This is a sample SMF manifest. If you want to actually use it, you'll
-  need to update this block comment and replace @@PREFIX@@ in the *generated*
-  file with the path to whereever this package gets installed.
- -->
-
-<service_bundle type="manifest" name="smartdc-ca-castashsvc">
-<service name="smartdc/application/volapi" type="service" version="1">
-
-<dependency name="network" grouping="require_all" restart_on="error" type="service">
-	<service_fmri value="svc:/network/physical" />
-</dependency>
-<dependency name="filesystem" grouping="require_all" restart_on="error" type="service">
-	<service_fmri value="svc:/system/filesystem/local" />
-</dependency>
-
-<exec_method
-    type="method"
-    name="start"
-    exec="@@NODE@@ --abort_on_uncaught_exception @@PREFIX@@/server.js &amp;"
-    timeout_seconds="30" />
-<exec_method type="method" name="stop" exec=":kill" timeout_seconds="30" />
-
-<instance name="default" enabled="true" />
-
-<stability value='Unstable' />
-
-<template>
-	<common_name><loctext xml:lang="C">SDC Volumes API</loctext></common_name>
-</template>
-
-</service>
-</service_bundle>
